src/js/components/SelectMultiple/SelectMultiple.js

```diff
@@ -152,6 +152,23 @@ const SelectMultiple = forwardRef(
       if (!search) setAllOptions(optionsProp);
     }, [optionsProp, search]);
 
+    useEffect(() => {
+      if (search && optionsProp && optionsProp.length > 0) {
+        const additionalOptions = [...allOptions];
+        optionsProp.forEach((i) =>
+          additionalOptions.some((j) => {
```

Comment:
Syntax question, could this also be written as `!additionalOptions.some(...) && additionalOptions.push(i)` -- or is the `null` return needed specifically?

Reply:
Good suggestion, the `null` return is not needed. Incorporated this suggestion in latest commit